discovery/stackit/server_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stackit
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"testing"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/common/promslog"
+	"github.com/stretchr/testify/require"
+)
+
+type serverSDTestSuite struct {
+	Mock *SDMock
+}
+
+func (s *serverSDTestSuite) SetupTest(t *testing.T) {
+	s.Mock = NewSDMock(t)
+	s.Mock.Setup()
+
+	s.Mock.HandleServers()
+}
+
+func TestServerSDRefresh(t *testing.T) {
+	for _, tc := range []struct {
+		name string
+		cfg  SDConfig
+	}{
+		{
+			name: "default with token",
+			cfg: func() SDConfig {
+				cfg := DefaultSDConfig
+				cfg.HTTPClientConfig.BearerToken = testToken
+
+				return cfg
+			}(),
+		},
+		{
+			name: "default with service account key",
```

Comment:
is this to make sure the token is retrieved?
maybe extract it into its own test, and keep this on for "right targets are built"?